Plugins~/Src/MeshSyncClientBlender/msblenContext.cpp

```diff
@@ -59,17 +65,29 @@ void msblenContext::NodeRecord::recordAnimation(msblenContext *_this) const {
 
 //----------------------------------------------------------------------------------------------------------------------
 
+static msblenContext* s_instance = nullptr;
+
 msblenContext& msblenContext::getInstance() {
-    static msblenContext s_instance;
-    return s_instance;
+    if (s_instance == nullptr) {
+        s_instance = new msblenContext();
+    }
+
+    return *s_instance;
 }
 
 void msblenContext::Destroy() {
+    if (!s_instance) {
+        return;
+    }
+
     m_texture_manager.clear();
     m_material_manager.clear();
 
     m_entities_state->clear();
     m_instances_state->clear();
+
+    delete s_instance;
+    s_instance = nullptr;
```

Comment:
Without this the context was not destroyed at the correct time, which caused the AsyncSceneSender to not be destroyed and that kept blender alive and didn't allow it to shut down because I have the long running socket request to unity to get changed data from it. With this change, the sender's destructor is called when the context is destroyed and the sender can cancel that request in its destructor.